akka-docs/src/test/java/jdocs/sharding/ClusterShardingTest.java

```diff
@@ -247,9 +256,11 @@ private void passivate() {
 
   static//#supervisor
   public class CounterSupervisor extends AbstractActor {
+    public static final String ShardingTypeName = "CounterSupervisor";
+    public static final Props props = Props.create(CounterSupervisor::new);
 
     private final ActorRef counter = getContext().actorOf(
-        Props.create(Counter.class), "theCounter");
+            Counter.propsFactory.apply("theCounter"), "theCounter");
```

Comment:
this would be weird, entities would get the id "theCounter"

The `CounterSupervisor` should also get its id from the factory parameter and pass that to `Counter.props(entityId)` here.

Reply:
and similar in the scala example

Reply:
That seemed odd to me too, but that's the way it was originally (`self.path.name` was "theCounter") so I didn't want to break anything. I'll change it.